File: webook-fe/types/comment.ts
// 对应后端 web.CommentVO（likeCnt/liked 由 core 聚合 interaction(biz="comment") 填入）
export interface Comment {
  id: number;
  user: CommentUser;
  content: string;
  rootId: number; // 根评论 id（一级评论=0）
  pid: number; // 直接父评论 id（一级=0）
  replyCnt: number; // 直接回复数
  likeCnt: number;
  liked: boolean; // 当前用户是否已点赞
  deleted: boolean; // 已删除占位（有子回复时保留，渲染「该评论已删除」）
  createdAt: number; // Unix 毫秒
  children?: Comment[]; // 一级评论携带的前 N 条回复预览（P0 走懒加载，通常为空）
}

export interface CommentUser {
  id: number;
  name: string;
}

// POST /comment/list — 一级评论分页
export interface ListCommentReq {
  articleId: number;
  sort: CommentSort;
  offset?: number;
  limit: number;
}

export type CommentSort = 'hot' | 'new';

// POST /comment/replies — 楼内回复懒加载
export interface GetRepliesReq {
  rootId: number;
  offset?: number;
  limit: number;
}

// POST /comment/create — 发表评论/回复
export interface CreateCommentReq {
  articleId: number;
  content: string;
  pid?: number; // 回复目标父评论；省略/0=一级评论
}

File: webook/internal/domain/comment.go
package domain

// CommentView 评论树视图——core 聚合 comment gRPC（评论树）+ interaction（likeCnt/liked）
// + user（昵称）后的领域对象，接入层再映射为 VO。
type CommentView struct {
	Id        int64
	User      CommentUser
	Content   string
	RootId    int64
	Pid       int64
	ReplyCnt  int64
	LikeCnt   int64
	Liked     bool
	CreatedAt int64
	Children  []CommentView
}

// CommentUser 评论者（core 按 uid 解析昵称填入；comment 服务只存 uid）。
type CommentUser struct {
	Id   int64
	Name string
}
